gosub-bindings/src/lib.rs

```diff
@@ -0,0 +1,121 @@
+use std::ffi::c_char;
+use std::ffi::CStr;
+use std::ptr;
+
+pub mod wrapper;
+
+use crate::wrapper::text::CTextNode;
+
+
+use gosub_engine::{
+    bytes::{CharIterator, Confidence, Encoding},
+    html5::parser::{
+        document::{Document, DocumentBuilder},
+        Html5Parser,
+    },
+    render_tree::{Node, NodeType, RenderTree, TreeIterator},
+};
+
+/// Initialize a render tree and return an owning pointer to it.
+/// If the HTML fails to parse, returns a NULL pointer.
+///
+/// # Safety
+/// Takes a read-only pointer owned from the C API representing the HTML source
+/// to build a render tree. DO NOT take ownership of this pointer in Rust or the
+/// universe might collapse.
+///
+/// Moves an owning pointer to the render_tree using Box::into_raw() to the C API.
+/// This pointer MUST be passed to gosub_render_tree_free() after usage for proper cleanup.
+#[no_mangle]
+pub unsafe extern "C" fn gosub_render_tree_init(html: *const c_char) -> *mut RenderTree {
+    let html_str: &str;
+    unsafe {
+        html_str = CStr::from_ptr(html).to_str().unwrap();
+    }
```

Comment:
```suggestion
    let html_str = unsafe {
        if let Ok(html_str) = CStr::from_ptr(html).to_str() {
            html_str
        } else {
            return ptr::null_mut();
        }
    };
```

Returning null to C instead of crashing the program when something fails is probably better

Reply:
yep, I agree